linked_list/linked_list.py

```diff
@@ -89,29 +170,95 @@ def visit(self):
     # Time Complexity: ?
     # Space Complexity: ?
     def reverse(self):
-        pass
-  
+        if self.head == None:
+            return 
+        elif self.head.next == None:
+            return
+        else:
+            current = self.head
+            reversed_link = None
+            while current.next != None:
+                if not reversed_link:
+                    next_node = current.next
+                    print(f"next_node: {next_node.value}")
+                    current.next = None
+                    reversed_link = current
+                    print(f"reversed_link: {reversed_link.value}")
+                    current = next_node
+                    print(f"current: {current.value}")
+                else:
+                    print(f"**********")
+                    print(f"hitting else block")
+                    next_node = current.next
+                    print(f"next_node: {next_node.value}")
+                    current.next = reversed_link
+                    reversed_link = current
+                    current = next_node
+                    print(f"current: {current.value}")

```

Comment:
Taking out `print` statements

```suggestion
            while current.next != None:
                if not reversed_link:
                    next_node = current.next
                    current.next = None
                    reversed_link = current
                    current = next_node
                else:
                    next_node = current.next
                    current.next = reversed_link
                    reversed_link = current
                    current = next_node
```